File: src-tauri/src/commands/connectivity.rs
use crate::models::result::CommandResult;
use crate::services::connectivity_service::{self, ConnectionConfigInput, ConnectionTestData};

#[tauri::command]
pub async fn test_connection(content: ConnectionConfigInput) -> CommandResult<ConnectionTestData> {
    match connectivity_service::test_connection(content).await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

File: src/main/ipc/connectivity.ts
import { ipcMain } from "electron";
import { IPC } from "../../shared/ipc-channels.js";
import { ok, err } from "../models/result.js";
import { toAppError } from "./utils.js";
import { testConnection } from "../services/connectivity-service.js";
import type { ConnectionConfigInput } from "../services/connectivity-service.js";

export function registerConnectivityHandlers(): void {
  ipcMain.handle(IPC.TEST_CONNECTION, async (_event, payload: ConnectionConfigInput) => {
    try { return ok(await testConnection(payload)); }
    catch (error) { return err(toAppError(error)); }
  });
}
